### PR TITLE
WOTD plugin for async-bot implementation

### PR DESCRIPTION
**Pull request by mega6382**
Implementations for WOTD from both dictionary.com and merriam-webster.com are added

### CODE REVIEW

File: src/Exception/InvalidDOMStructure.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+namespace AsyncBot\Plugin\WordOfTheDay\Exception;
+
+
+class InvalidDOMStructure extends Exception
+{
+    public function __construct()
+    {
+        parent::__construct('Unexpected html structure');
+    }
+}
```
Comment (PeeHaa) [inline src/Exception/InvalidDOMStructure.php]: Line break

File: src/Parser/GetFromDictionaryDotComResult.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        $url        = 'http://www.dictionary.com/browse/' . str_replace(' ', '-', $word);
+        $definition = trim($definitionNodes->item(0)->textContent);
+
+        if (strlen($word) === 0) {
+            throw new WotdNotFound();
+        }
+
+        return new Wotd($word, $url, $definition);
+    }
+}
```
Comment (PeeHaa) [inline src/Parser/GetFromDictionaryDotComResult.php]: Line break

File: src/Parser/GetFromMerriamWebsterResult.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        $url        = 'https://www.merriam-webster.com/dictionary/' . str_replace(' ', '-', $word);
+        $definition = $definitionNodes->item(0)->getElementsByTagName('p')->item(0)->textContent;
+
+        if (strlen($word) === 0) {
+            throw new WotdNotFound();
+        }
+
+        return new Wotd($word, $url, $definition);
+    }
+}
```
Comment (PeeHaa) [inline src/Parser/GetFromMerriamWebsterResult.php]: Line break

File: src/Exception/InvalidDOMStructure.php [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<?php
+declare(strict_types=1);
```
Comment (PeeHaa) [inline src/Exception/InvalidDOMStructure.php]: Move to first line

File: src/Parser/GetFromDictionaryDotComResult.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
+<?php
+declare(strict_types=1);
```
Comment (PeeHaa) [inline src/Parser/GetFromDictionaryDotComResult.php]: Move to first line

File: src/Parser/GetFromMerriamWebsterResult.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
+<?php
+declare(strict_types=1);
```
Comment (PeeHaa) [inline src/Parser/GetFromMerriamWebsterResult.php]: Move to first line

File: src/Retriever/GetFromDictionaryDotCom.php [hunk 7]
```diff
@@ -0,0 +1,36 @@
+<?php
+declare(strict_types=1);
```
Comment (PeeHaa) [inline src/Retriever/GetFromDictionaryDotCom.php]: Move to first line

File: src/Retriever/GetFromMerriamWebster.php [hunk 8]
```diff
@@ -0,0 +1,35 @@
+<?php
+declare(strict_types=1);
```
Comment (PeeHaa) [inline src/Retriever/GetFromMerriamWebster.php]: Move to first line

File: src/Retriever/GetFromMerriamWebster.php [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+     */
+    public function retrieve(): Promise
+    {
+        return call(function () {
+            return (new GetFromMerriamWebsterResult())->parse(
+            /** @var Response $response */
+                yield $this->httpClient->requestHtml('https://www.merriam-webster.com/word-of-the-day'));
+        });
+    }
+}
```
Comment (PeeHaa) [inline src/Retriever/GetFromMerriamWebster.php]: Line break

File: src/valueObject/Result/Wotd.php [hunk 10]
```diff
@@ -0,0 +1,36 @@
+<?php
+declare(strict_types=1);
```
Comment (PeeHaa) [inline src/valueObject/Result/Wotd.php]: Move to first line

File: src/valueObject/Result/Wotd.php [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    public function getUrl(): string
+    {
+        return $this->url;
+    }
+
+    public function getDefinition(): string
+    {
+        return $this->definition;
+    }
+}
```
Comment (PeeHaa) [inline src/valueObject/Result/Wotd.php]: Line break

**Review by PeeHaa — CHANGES_REQUESTED** [review]
Can you also add tests please? <3

**Comment by mega6382** [comment]
> Can you also add tests please? <3

I will do that in another PR. In the meanwhile I will fix the issues that you have mentioned.

**Comment by mega6382** [comment]
@PeeHaa done

**Review by PeeHaa — APPROVED** [review]

**Pull request merged**